### ISSUE TITLE
Example of VanillaLSTMBuilder.set_dropout_masks?

### ISSUE THREAD

**Issue by johntzwei** [clab/dynet#844]
Hi all,

I am currently manually minibatching my code for a seq2seq model. I am using masking to batch together my examples. I understand that for the output, I can multiply each output token's loss by the mask value, and have a masked output.

How do I mask the input? I think that set_dropout_masks seems to be the solution to my problem, but I'm not sure how to use it. Any help would be greatly appreciated! Thanks!

**Comment by pmichel31415** [comment]
no set_dropout_mask is for dropout. You would have to to create an expression for your mask and do cmult with the loss

**Comment by johntzwei** [comment]
Is it possible to mask the input?

For example, if I had the sentence 'I use dynet . <EOS> <mask>', and I batched it together with other sentences into X_batch, and used: LSTMState.add_inputs(X_batch), is it possible in Dynet to skip timesteps?

**Comment by neubig** [comment]
No, it's not, but this would be a good feature to have (I was wanting this myself). I'll mark it as an enhancement.

**Comment by liesun1994** [comment]
I think it works well with   start_new_sequence() 
if you want to use an mask ,  start_new_sequence() is a good choice. 
Examples 
    `Expression states = lstm_builder.add_inputs(expr);`
  ` vector<Expression> c&s = lstm_builder.final_s();  // here c&s means two output in lstm.`
  ` Expression c = c&s[0];`
   `Expression now_c = c*mask+ c_pre(1-mask);  // here mask is an Expression in short `
  ` Expression now_s = states*mask + states_pre* (1-mask);`
   `lstm_builder->start_new_sequence(<now_c, now_s>);`

then the states updated with mask you provided .
It a good choice if it could be done on top of dynet.
Aha

**Comment by johntzwei** [comment]
Hi, sorry to bother, but will this be updated soon? Thanks!

**Comment by liesun1994** [comment]
I think it will not be updated soon. I tried the same thing with you before, if you want to used masked_input, you can shift your input and change the output states you wanted (of course with mask). Just as I mentioned before .  The function start_new_sequence() actually changes the states of LSTM.

**Comment by liesun1994** [comment]
@johntzwei Aha, forget to look into your questions. Sry
